### PR TITLE
AGENT-51: Update config_example.sh to be more documented

### PR DESCRIPTION
**Pull request by lranjbar**
This updates the formatting of config_example.sh. I tried to keep the content the same as much as possible. I have made some Ansible that will generate a config_$USER.sh for you and this is what the generated config_$USER.sh looks like.

@zaneb @pawanpinjarkar @andfasano @rwsu @bfournie

### CODE REVIEW

File: config_example.sh [hunk 1]
```diff
@@ -1,287 +1,743 @@
 #!/bin/bash
 
+####################################################################################################
+## Development Settings
+##
+
+################################################################################
```
[the remark this answers is not in the document]
Comment (rwsu) [reply config_example.sh:7]: I concur. I find them a bit distracting.

File: config_example.sh [hunk 2]
```diff
@@ -1,287 +1,743 @@
...
 # You can get this token from https://console-openshift-console.apps.ci.l2s4.p1.openshiftapps.com/ by
 # clicking on your name in the top right corner and coping the login
 # command (the token is part of the command)
 set +x
 export CI_TOKEN=''
 set -x
+################################################################################
 
+################################################################################
+# CI_SERVER -
```
Comment (rwsu) [inline config_example.sh:18]: I like the variable name followed by description format change.

File: config_example.sh [hunk 3]
```diff
@@ -1,287 +1,743 @@
...
+#
+# metallb container location (optional)
+#export METALLB_IMAGE_BASE=
+#export METALLB_IMAGE_TAG=
+#
+################################################################################
 
-# Provisioning interface on the helper ndoe
+####################################################################################################
+## Network Settings
```
Comment (rwsu) [inline config_example.sh:262]: The subsections are a nice way to see the different types of configuration options.

**Review by rwsu — COMMENTED** [review]
I like the change overall. Like Zane suggested I would suggest removing the long "###" bars for variables and leave them for the sections.

**Comment by zaneb** [comment]
/lgtm
Thanks Lisa!

**Review by rwsu — COMMENTED** [review]
/lgtm

Nice improvement.

File: config_example.sh [hunk 4]
```diff
@@ -1,287 +1,545 @@
...
+# DNS_VIP (Defaults to v6)
+#
+# v4 Default:
 #export DNS_VIP="11.0.0.2"
+#
+# v6 Default:
+#export DNS_VIP="fd2e:6f44:5dd8:c956::2"
 
-# Network type
+# Network type (Defaults to v6)
```
Comment (hardys) [inline config_example.sh:242]: The `defaults to v6` here and elsewhere is slightly misleading - the values for `NETWORK_TYPE`, and subnet/VIP values are [defaulted depending on the value](https://github.com/openshift-metal3/dev-scripts/blob/master/network.sh#L104..L130) of `IP_STACK`

Comment (hardys) [inline config_example.sh:242]: The `defaults to v6` here and elsewhere is slightly misleading - the values for `NETWORK_TYPE`, and subnet/VIP values are [defaulted depending on the value](https://github.com/openshift-metal3/dev-scripts/blob/master/network.sh#L104..L130) of `IP_STACK`

**Comment by hardys** [comment]
One small nit mentioned which would be nice to fix, otherwise lgtm, thanks!

**Comment by hardys** [comment]
One small nit mentioned which would be nice to fix, otherwise lgtm, thanks!

File: config_example.sh (hunk 4 above)
Comment (lranjbar) [reply config_example.sh:242]: That's true these values are based off `IP_STACK` and `HOST_IP_STACK`. 🤔 I will update the wording.

**Comment by zaneb** [comment]
@lranjbar stop checking GitHub notifications when you are on holiday.

/approve

**Pull request merged**